azext_iot/common/base_discovery.py

```diff
@@ -339,3 +358,8 @@ def _build_target(self, resource, policy, key_type=None, **kwargs):
         """Returns a dictionary representing the resource connection string parts to
         be used by the IoT extension."""
         pass
+
+    @abstractmethod
+    def _build_target_from_name(cls, hostname, resource_group_name):
```

Comment:
Why have `resource_group_name` parameter? In this use case, if someone provides their hub hostname no rg is necessary.

Reply:
there is a case with root authority set but that is the only case so far. I will change it to deal with it in a similar fashion as dps registration with id scope